java/arcs/sdk/Handle.kt

```diff
@@ -37,12 +37,18 @@ interface WritableSingleton<T : Entity> : Handle {
 interface ReadWriteSingleton<T : Entity> : ReadableSingleton<T>, WritableSingleton<T>
 
 /** A collection handle with read access. */
-interface ReadableCollection<T : Entity> : Handle, Iterable<T> {
+interface ReadableCollection<T : Entity> : Handle {
     /** The number of elements in the collection. */
     val size: Int
 
     /** Returns true if the collection is empty. */
     fun isEmpty(): Boolean
+
+    /** Returns an Iterator */
+    fun iterator(): Iterator<T>
```

Comment:
This should not be here. Replace usages of iterator() to fetchAll().iterator()

Reply:
iterator() is a method that should be on iterables, but we're not an iterable anymore.

Reply:
Updated